TransIT.DAL/Repositories/ImplementedRepositories/PartRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LinqKit;
+using Microsoft.EntityFrameworkCore;
+using TransIT.DAL.Models;
+using TransIT.DAL.Models.Entities;
+using TransIT.DAL.Repositories.InterfacesRepositories;
+
+namespace TransIT.DAL.Repositories.ImplementedRepositories
+{
+    public class PartRepository : BaseRepository<Part>, IPartRepository
+    {
+        public PartRepository(TransITDBContext context)
+               : base(context)
+        {
+        }
+
+        public override Task<IQueryable<Part>> SearchExpressionAsync(IEnumerable<string> strs)
+        {
+            var predicate = PredicateBuilder.New<Part>();
+
+            foreach (string keyword in strs)
+            {
+                predicate = predicate.And(entity =>
+                   EF.Functions.Like(entity.Name, '%' + keyword + '%')
+                   );
+            }
+
+            return Task.FromResult(
```

Comment:
+ to the previous comment, this logic also is very common I think, so, could it be moved to BaseRepository?